src/main/java/org/gridsuite/study/server/service/SingleLineDiagramService.java

```diff
@@ -70,13 +71,9 @@ public byte[] getVoltageLevelSvg(UUID networkUuid, String variantId, String volt
                 .queryParam(QUERY_PARAM_USE_NAME, diagramParameters.isUseName())
                 .queryParam(QUERY_PARAM_CENTER_LABEL, diagramParameters.isLabelCentered())
                 .queryParam(QUERY_PARAM_DIAGONAL_LABEL, diagramParameters.isDiagonalLabel())
-                .queryParam(QUERY_PARAM_TOPOLOGICAL_COLORING, diagramParameters.isTopologicalColoring());
-        if (diagramParameters.getComponentLibrary() != null) {
-            uriComponentsBuilder.queryParam(QUERY_PARAM_COMPONENT_LIBRARY, diagramParameters.getComponentLibrary());
-        }
-        if (!StringUtils.isBlank(variantId)) {
-            uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
-        }
+                .queryParam(QUERY_PARAM_TOPOLOGICAL_COLORING, diagramParameters.isTopologicalColoring())
+                .queryParam(LANGUAGE, diagramParameters.getLanguage());
+        addParameters(diagramParameters, uriComponentsBuilder, variantId);
```

Comment:
why not setting all parameters with addParameters method (and not only the component library and variant id) ?

Reply:
if we add it in addParameters we will lose the order of parameters and it will impact our TU because the URL are writen in text and not built. that's why I externalized only parameters added with conditions. We have to change it in another ticket